readme.md

```diff
@@ -200,119 +107,244 @@ $ adb forward tcp:9222 localabstract:chrome_devtools_remote
 $ lighthouse --disable-device-emulation --disable-cpu-throttling https://mysite.com
 ```
 
-## Tests
+## Using programmatically
 
-Some basic unit tests forked are in `/test` and run via mocha. ESLint also checks for style violations.
+The example below shows how to setup and run Lighthouse programmatically as a Node module.
 
-```sh
-# lint and test all files
-npm test
+```javascript
+// Install dependencies: yarn install --dev lighthouse yargs
 
-# watch for file changes and run tests
-npm run watch
+'use strict';
 
-## run linting and unit tests separately
-npm run lint
-npm run unit
+const lighthouse = require('lighthouse');
+const ChromeLauncher = require('lighthouse/lighthouse-cli/chrome-launcher.js').ChromeLauncher;
+const Printer = require('lighthouse/lighthouse-cli/printer');
+const yargs = require('yargs');
+
+// Also could use your own custom config.
+const PERF_CONFIG = require('lighthouse/lighthouse-core/config/perf.json');
+// const DEFAULT_CONFIG = require('lighthouse/lighthouse-core/config/default.json');
+
+const _SIGINT = 'SIGINT';
```

Comment:
Doesn't this seem like overkill? Handling command line args and SIGINT seems out of scope for a readme example